### PR TITLE
Give executor to cache instead of string

### PR DESCRIPTION
**Pull request by kingherc**
Relates ES-8155

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java [hunk 1]
```diff
@@ -568,6 +568,7 @@ public ExecutorService generic() {
     public ExecutorService executor(String name) {
         final ExecutorHolder holder = executors.get(name);
         if (holder == null) {
+            logger.warn("IRAKLIS all executors: {}", executors);
```
Comment (arteam) [inline server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java:571]: Nit: I assume it's an unintended change.

**Review by arteam — APPROVED** [review]
LGTM

**Review by tlrx — APPROVED** [review]
LGTM

**Review by idegtiarenko — APPROVED** [review]

**Review by henningandersen — APPROVED** [review]
LGTM.

**Review by volodk85 — APPROVED** [review]
LGTM

**Pull request merged**